src/dotnet/updater.py

```diff
@@ -317,25 +340,11 @@ def _guess_version_from_pkglist(self, pkg_list: list[RpmPackage]) -> str | None:
 
     def generate_custom_end(self) -> None:
         assert self.package_list
-        if not DotNetBCI._base:
-            DotNetBCI._base = dnf.Base()
-            DotNetBCI._base.repos.add_new_repo(
-                repoid="packages-microsoft-com-prod",
-                conf=DotNetBCI._base.conf,
-                baseurl=(MS_REPO_BASEURL,),
-            )
-            DotNetBCI._base.fill_sack()
-
-        pkgs = self._fetch_packages()
-
-        if new_version := self._guess_version_from_pkglist(pkgs):
-            assert not self.additional_versions, f"additional_versions property must be unset, but got {self.additional_versions}"
-            self.additional_versions = [new_version]
```

Comment:
This is a very costly operation, moving it to `__post_init__()` will cause it to get executed on every import. This will also slow down the bot considerably (just calling `scratch_build_bot --help` takes five times longer on this branch). I'd greatly appreciate if we could keep this costly part here and not move it into the constructor.

Reply:
It's not cleanly possible. DevelopmentContainer's __post_init__ barfs when self.version is None, so it has to be determined at init time. 

The alternative is that we set self.version to some random garbage value and fix it up later. Which I didn't go for because I think that'll just be causing hard to trace subtle bugs later on. 

The 3rd alternative is that we make update-versions determine the dotnet package versions and store it in package_version.json instead.

Reply:
which of the options do you prefer? don't have time to implement all of them.

Reply:
I would go with `package_version.json` if it is easy to implement.

Reply:
I wasn't able to get this done easily, the package version logic is too tied up with OBS. so I added a `prepare_generation` hook that can be used to generate expensive things outside initalization and before generating the templates.